Document steps for updating prod screenshots in tests

mon4our opened the pull request:
Updated the acceptance tests wiki with steps on how to update screenshots

mon4our committed:
Document steps for updating prod screenshots in tests

mon4our commented:
@seanlip PTAL, thanks!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
 On the other hand, if the screenshot fails locally (in desktop environment), the screenshot `teachPage-diff.png` will be generated and stored inside a new folder `diff-snapshots` under `logged-out-user/dev-desktop-screenshots`. 
 
+### Adding Prod screenshots for Acceptance Tests
+On making changes that affect a user journey tested through the acceptance tests or introduce a new feature, a contributor needs to update/add screenshots supporting their changes.
```
On --> When

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
 
+### Adding Prod screenshots for Acceptance Tests
+On making changes that affect a user journey tested through the acceptance tests or introduce a new feature, a contributor needs to update/add screenshots supporting their changes.
+It has been noted that screenshots obtained by running the Acceptance tests locally using the prod flag do not match those captured by the CI. To overcome this, you can commit any image in the prod screenshot folder and follow these steps:
```
Drop the first sentence, replace with: "These screenshots should be obtained from the CI (not local) run, so that the environment matches future runs." Update the second sentence to follow on from that.

In the second sentence, what do you mean by "commit any image in the prod screenshot folder"?

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+### Adding Prod screenshots for Acceptance Tests
+On making changes that affect a user journey tested through the acceptance tests or introduce a new feature, a contributor needs to update/add screenshots supporting their changes.
+It has been noted that screenshots obtained by running the Acceptance tests locally using the prod flag do not match those captured by the CI. To overcome this, you can commit any image in the prod screenshot folder and follow these steps:
+1. Go to the failing run and click on the 'uploading diff screenshots as an artifact' section:
```
Capitalize "uploading"

"click on the" --> "open the"

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+1. Go to the failing run and click on the 'uploading diff screenshots as an artifact' section:
+  <img width="349" height="35" alt="image" src="https://github.com/user-attachments/assets/827d8557-3346-443a-aaf8-1fd0168a3f24" />
+
+2. Then download the artifact from the artifact download url:
```
Highlight that URL in the image.

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+2. Then download the artifact from the artifact download url:
+  <img width="1221" height="301" alt="image" src="https://github.com/user-attachments/assets/c2caa55f-b4f9-40cc-a47a-7ddc2de9a479" />
+
+3. Extract the contents of the artifact. There will be a folder with an image. The image will be like this:
```
be like this --> look like this

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+6. Choose Split into pieces:
+<img width="699" height="191" alt="image" src="https://github.com/user-attachments/assets/207fb140-740a-4cf1-bc12-8823a9b1e3b8" />
+
+7. Set rows as 3 and columns as 1:
```
Strange ... this is definitely correct, right? I would have thought the terminology would be splitting the image into 3 columns, so just checking.

mon4our replied:
Yep, the site splits the images the way we want only when we change the rows value. I found it a bit weird as well.

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+7. Set rows as 3 and columns as 1:
+<img width="699" height="191" alt="image" src="https://github.com/user-attachments/assets/19ae5e5e-1a90-4b0d-bb88-59e75b3d53ce" />
+
+8. Click on split image 
```
split image --> "SPLIT IMAGE" (follow the screenshot)

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+
+10. Right click on the rightmost one and use 'Save Image as...'. Choose any random name.
+
+11. Navigate to where you have saved the Oppia repo on your local machine.
```
Combine this with the next step.

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+
+12. Go to oppia/core/tests/puppeteer-acceptance-tests/specs
+
+13. The spec will be mentioned in the artifact folder you downloaded:
```
In which step did I download this "artifact folder"? Maybe refer to it.

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+
+<img width="368" height="64" alt="Screenshot from 2025-08-20 16-37-52" src="https://github.com/user-attachments/assets/d1a45271-c1f1-49d9-8874-f5ac95d37dce" />
+
+(Its curriculum-admin in this case)
```
What does "Its" refer to?

Also note that "its" is the possessive, and "it's" means "it is".

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+
+(Its curriculum-admin in this case)
+
+14. Navigate to the spec
```
This is unclear, please specify what you mean by "spec". How do they figure out which spec to navigate to?

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+15. Check if the failure was for desktop or mobile based on the screenshot or the subfolder in the artifact:
+<img width="102" height="127" alt="image" src="https://github.com/user-attachments/assets/c1b902ee-5ccb-4218-a247-d4e19da9d3ed" />
+
+16. Go to the prod-desktop-screenshots/prod-mobile-screenshots folder depending on your failure
```
Use "or" instead of "/" to avoid ambiguity, since a slash is a folder name separator too

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+
+16. Go to the prod-desktop-screenshots/prod-mobile-screenshots folder depending on your failure
+
+17. Paste the image we split.
```
Refer to the step number where the splitting action was taken

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+
+17. Paste the image we split.
+
+18. Check the name of the image from the artifact. It will be {screenshot_name}-diff.png
```
I think you can combine steps 18-21 into something along the lines of "replace the old screenshot with the new one"

mon4our replied:
Done!

seanlip commented on Acceptance-Tests.md:
```diff
@@ -247,6 +247,67 @@ On CI, we run all the acceptance tests in production mode, so the screenshots in
+
+20. Rename the pasted image to {screenshot_name}-snap.png
+
+21. Check that the correct image got replaced, commit and push your changes!
```
"Commit and push your changes" should be a separate step

Maybe at the end you can also ask them to do a self-review of the PR to verify that the correct images are used.

mon4our replied:
Done!

seanlip reviewed (changes requested):
Thanks @mon4our for doing this! Left a few notes, PTAL.

seanlip commented:
@mon4our Just to check, any reason this isn't wrapped up yet? It seems like it'd be quite useful to have on the wiki.

mon4our committed:
Revise instructions for adding prod screenshots

Updated instructions for adding production screenshots in acceptance tests to clarify the process and improve readability.

mon4our commented:
@seanlip Made the requested changes. PTAL, thanks!

seanlip reviewed (approved):
Looks great. Thanks @mon4our!
